x/incentives/keeper/distribute.go

```diff
@@ -441,7 +480,10 @@ func (k Keeper) Distribute(ctx sdk.Context, gauges []types.Gauge) (sdk.Coins, er
 			ctx.Logger().Debug("distributeSyntheticInternal, gauge id %d, %d", "module", types.ModuleName, "gaugeId", gauge.Id, "height", ctx.BlockHeight())
 			gaugeDistributedCoins, err = k.distributeSyntheticInternal(ctx, gauge, filteredLocks, &distrInfo)
 		} else {
-			gaugeDistributedCoins, err = k.distributeInternal(ctx, gauge, filteredLocks, &distrInfo)
+			// Donot distribue if LockQueryType = Group, because if we distribute here we will be double distributing.
+			if gauge.DistributeTo.LockQueryType != lockuptypes.ByGroup {
```

Comment:
I believe skipping in this way might lead to `totalDistributedCoins` just double counting the previous loop's `gaugeDistributedCoins` (if this is the case, let's make sure that tests catch this in the future).

The more appropriate logic seems to be `if [same condition], continue` and then leave the `distributeInternal` unchanged

Reply:
oh yea! we def want to skip the rest of the operation post this call

Reply:
added: https://github.com/osmosis-labs/osmosis/pull/6085/files#diff-620e494b71d3ef447092659ad8b99835a3088760bb0924956007e88b3745f5bcR505